backend/inspirehep/records/marshmallow/literature/ui.py

```diff
@@ -264,6 +267,16 @@ def get_datasets(self, data):
                 )
         return dataset_links or missing
 
+    def get_datasets_from_data_links(self, data):
+        dataset_links = []
+        all_links = get_value(data, "data", [])
+        for link in all_links:
+            data_link = link.get("record").get("$ref")
+            control_number = data_link.split("/")[-1]
+            data_ui_link = data_link.replace("/api", "")
```

Comment:
from inspire_dojson.utils import get_recid_from_ref
```suggestion
            data_control_number = get_recid_from_ref(..)
            ....
   
```